### PR TITLE
WIP: add env var for configuration

### PR DESCRIPTION
**Pull request by akoserwal**
## Describe your changes
Add ENV variables in config.yaml 

- ...

## Ticket reference (if applicable)
Fixes #

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

**Comment by akoserwal** [comment]
https://github.com/go-kratos/kratos/issues/3198#issuecomment-1943440755

File: configs/config.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
   grpc:
-    addr: 0.0.0.0:9000
+    addr: "${GRPCADDR:0.0.0.0:9000}"
     timeout: 1s
 data:
   spiceDb:
     useTLS: false
-    endpoint: spicedb:50051
-    token: "${SPICEDB_PRESHARED:foobar}"
+    endpoint: "${ENDPOINT:spicedb:50051}"
```
Comment (merlante) [inline configs/config.yaml:11]: Has this config been tested with `make rebac`, etc. It looks odd, but if it works it works. :)

**Comment by shenqidebaozi** [comment]
@akoserwal @merlante 

I'm glad to see that RedHat is using Kratos. What are the reasons for using Kratos? We would like to know why users choose to use Kratos to help improve the project continuously.

Can you also add a use case, https://github.com/go-kratos/kratos/issues/969

If you have any questions, suggestions or opinions while using it, please feel free to give me feedback.

**Comment by akoserwal** [comment]
> @akoserwal @merlante
> 
> I'm glad to see that RedHat is using Kratos. What are the reasons for using Kratos? We would like to know why users choose to use Kratos to help improve the project continuously.
> 
> Can you also add a use case, [go-kratos/kratos#969](https://github.com/go-kratos/kratos/issues/969)
> 
> If you have any questions, suggestions or opinions while using it, please feel free to give me feedback.

Hi @shenqidebaozi thank you for reaching out. 

We are evaluating Kratos for a POC project. Kratos enabled us to develop a proto-to-open API spec workflow and build APIs quickly. 

Suggestions: The discord go-kratos forum isn't active much. Is there any other forum where I can engage for doubts/discussions? 

As we go along with the development, I'll definitely identify areas to improve and where we see an opportunity to contribute.

**Comment by shenqidebaozi** [comment]
@akoserwal Due to certain network restrictions, Discord cannot be directly accessed within China. At present, our main focus is on maintaining the community on WeChat. But you can also add my Discord(baozi6562) for communication, and I am also considering how to improve the issue you encountered when using config env.

Since the values of environment variables are all strings, it is necessary to convert them to the expected target type when using them. I am considering how to implement this feature. Do you have any ideas?

**Comment by akoserwal** [comment]
> Since the values of environment variables are all strings, it is necessary to convert them to the expected target type when using them. I am considering how to implement this feature.



> @akoserwal Due to certain network restrictions, Discord cannot be directly accessed within China. At present, our main focus is on maintaining the community on WeChat. But you can also add my Discord(baozi6562) for communication, and I am also considering how to improve the issue you encountered when using config env.
> 
> Since the values of environment variables are all strings, it is necessary to convert them to the expected target type when using them. I am considering how to implement this feature. Do you have any ideas?

One way would be to define the `type: int` along the fields in config.yaml and, post-reading the environment variable, do the conversion like this:
```// For int environment variables
envVar := os.Getenv("ENV_VAR_INT")
var1, err := strconv.Atoi(envVar)
```

Or allow users to read all environment variables as strings, and then they can do the conversion.

**Pull request closed without merging**